test/test_torch.py

```diff
@@ -8079,6 +8079,40 @@ def test_complex_flip(self, device, dtype):
                 np_fn = partial(np.flip, axis=flip_dim)
                 self._np_compare_func((torch_fn, np_fn), data, device, dtype)
 
+    def _test_fliplr_flipud(self, funcs, min_dim, max_dim, device, dtype):
+        for dim in range(min_dim, max_dim + 1):
+            shape = self._rand_shape(dim, 5, 10)
+            # Randomly scale the input
+            data = (torch.randn(*shape).to(dtype) * random.randint(50, 100)).tolist()
+            # Use _np_compare_func as it copies the output of np_func,
+            # which takes care of negative strides if present.
+            self._np_compare_func(funcs, data, device, dtype)
+
+    @dtypes(torch.float, torch.double, torch.int16, torch.int32, torch.int64, torch.cfloat, torch.cdouble)
+    @unittest.skipIf(not TEST_NUMPY, "Numpy not found")
+    def test_fliplr(self, device, dtype):
+        funcs = (torch.fliplr, np.fliplr)
+        self._test_fliplr_flipud(funcs, 2, 4, device, dtype)
+
+    @dtypes(torch.float, torch.double, torch.int16, torch.int32, torch.int64, torch.cfloat, torch.cdouble)
+    def test_fliplr_invalid(self, device, dtype):
+        x = torch.randn(42).to(dtype)
+        with self.assertRaisesRegex(RuntimeError, "Input must be >= 2-d."):
+            torch.fliplr(x)
+        with self.assertRaisesRegex(RuntimeError, "Input must be >= 2-d."):
+            torch.fliplr(torch.tensor(42))
+
+    @dtypes(torch.float, torch.double, torch.int16, torch.int32, torch.int64, torch.cfloat, torch.cdouble)
+    @unittest.skipIf(not TEST_NUMPY, "Numpy not found")
+    def test_flipud(self, device, dtype):
+        funcs = (torch.flipud, np.flipud)
+        self._test_fliplr_flipud(funcs, 1, 4, device, dtype)
+
+    @dtypes(torch.float, torch.double, torch.int16, torch.int32, torch.int64, torch.cfloat, torch.cdouble)
```

Comment:
It's great how thorough these are, but it's probably OK just to test float32, int64, and c64.

Reply:
Updated.